packages/code-du-travail-ui/src/Toast/index.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled, { css } from "styled-components";
+import IconWarning from "react-feather/dist/icons/alert-triangle";
+import IconInfo from "react-feather/dist/icons/info";
+import IconSuccess from "react-feather/dist/icons/check";
+import IconClose from "react-feather/dist/icons/x";
+import ToggleButton from "../ToggleButton";
+import { box, colors, fonts, spacing } from "../theme";
+import { fromTop, fromRight, fromBottom, fromLeft } from "../keyframes";
+
+class Toast extends Component {
+  constructor(props) {
+    super(props);
+    this.timer = null;
+  }
+
+  componentDidMount() {
+    const { timeout } = this.props;
+
+    if (timeout) {
+      this.timer = setTimeout(() => {
+        this.props.onRemove();
+      }, timeout);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+  }
+
+  render() {
+    const { onRemove, type, className } = this.props;
+    let Icon = IconInfo;
+    if (type === "warning") Icon = IconWarning;
+    if (type === "success") Icon = IconSuccess;
+
+    return (
+      <div className={className}>
+        <IconWrapper type={type}>
+          <Icon />
+        </IconWrapper>
+        <Content role="alert">{this.props.children}</Content>
+        {onRemove ? (
+          <ButtonWrapper>
+            <ToggleButton variant="icon" onClick={onRemove}>
+              <IconClose />
+            </ToggleButton>
+          </ButtonWrapper>
+        ) : null}
+      </div>
+    );
+  }
+}
+
+Toast.propTypes = {
+  className: PropTypes.string,
+  type: PropTypes.oneOf(["warning", "success", "info"]),
+  wide: PropTypes.bool,
+  shadow: PropTypes.bool,
+  animate: PropTypes.oneOf([
+    "from-top",
+    "from-right",
+    "from-bottom",
+    "from-left"
+  ]),
+  timeout: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
```

Comment:
why `PropTypes.oneOfType([PropTypes.bool, PropTypes.number])` and not just `PropTypes.number`

Reply:
Humm I have this in the code `    if (timeout) {`  but I guess that if I leave it undefined by default instead of false it would also do the trick ?

Reply:
yes